types/decimal.go

```diff
@@ -318,30 +318,74 @@ func (d Dec) QuoInt64(i int64) Dec {
 	return Dec{mul}
 }
 
-// ApproxSqrt returns an approximate sqrt estimation using Newton's method to
-// compute square roots x=√d for d > 0. The algorithm starts with some guess and
+// ApproxRoot returns an approximate estimation of a Dec's positive real nth root
+// using Newton's method (where n is positive). The algorithm starts with some guess and
 // computes the sequence of improved guesses until an answer converges to an
-// approximate answer. It returns -(sqrt(abs(d)) if input is negative.
-func (d Dec) ApproxSqrt() Dec {
+// approximate answer.  It returns -(sqrt(abs(d)) if input is negative.
```

Comment:
> It returns -(sqrt(abs(d)) if input is negative.

this is not the case anymore

Reply:
Oh right, it still does the negative return, but it's not sqrt.